tests/test_multi_res_resource.py

```diff
@@ -0,0 +1,154 @@
+        ti = hr_res.time_index
+        meta = hr_res.meta
+        lr_data = [hr_res[dset] for dset in lr_dsets]
+        lr_attrs = hr_res.attrs
+        lr_chunks = hr_res.chunks
+        lr_dtypes = hr_res.dtypes
+
+    t_slice = slice(None, None, 12)
+    s_slice = slice(None, None, 10)
+    lr_ti = ti[t_slice]
+    lr_meta = meta.iloc[s_slice]
+    lr_data = [d[t_slice, s_slice] for d in lr_data]
+    lr_shapes = {d: (len(lr_ti), len(lr_meta)) for d in lr_dsets}
+
+    if interp_hh:
+        new_dsets = []
+        for i, dset in enumerate(lr_dsets):
+            new = dset.replace('100m', '20m')
+            new_dsets.append(new)
+            lr_dtypes[new] = lr_dtypes[dset]
+            lr_shapes[new] = lr_shapes[dset]
+            lr_attrs[new] = lr_attrs[dset]
+            lr_chunks[new] = lr_chunks[dset]
+            lr_data.append(lr_data[i].copy() * 1.1)
+        lr_dsets += new_dsets
+
+    Outputs.init_h5(fp_lr, lr_dsets, lr_shapes, lr_attrs, lr_chunks,
+                    lr_dtypes, lr_meta, lr_ti)
+    for name, arr in zip(lr_dsets, lr_data):
+        Outputs.add_dataset(fp_lr, name, arr, lr_dtypes[name],
+                            attrs=lr_attrs[name], chunks=lr_chunks[name])
+
+    with h5py.File(fp_hr, 'a') as f:
+        for dset in lr_dsets:
+            if dset in f:
+                del f[dset]
+
+    lr_res = WindResource(fp_lr)
+    hr_res = WindResource(fp_hr)
+    mrr = MultiResolutionResource(fp_hr, fp_lr, handler_class=WindResource)
+    assert len(mrr._nn_map) == len(hr_res.meta)
+    assert all(np.isin(mrr._nn_map, np.arange(len(lr_res.meta))))
+
+    assert all(d in mrr.dsets for d in all_dsets)
+    assert all(d in mrr.shapes for d in all_dsets)
+    assert all(d in mrr.scale_factors for d in all_dsets)
+    assert all(d in mrr.attrs for d in all_dsets)
+    assert all(d in mrr.attrs for d in all_dsets)
+
+    return fp_hr, fp_lr
+
+
+def test_mrr_indexing():
+    """Test data indexing with the multi resolution resource handler."""
+    with tempfile.TemporaryDirectory() as td:
+        fp_hr, fp_lr = make_multi_res_files(td)
+
+        lr_res = WindResource(fp_lr)
+        hr_res = WindResource(fp_hr)
+        mrr = MultiResolutionResource(fp_hr, fp_lr, handler_class=WindResource)
+        tree = KDTree(lr_res.coordinates)
+
+        dsets = ('pressure_100m', 'temperature_100m')
+        gids_hr = (0, 9, [1], [0, 3, 4], slice(None), slice(3, 15, 3))
+        for dset in dsets:
+            lr_gids = tree.query(hr_res.coordinates)[1]
+            lr_data = lr_res[dset, :, lr_gids]
+            hr_data = mrr[dset]
+            assert np.allclose(lr_data, hr_data[::12])
+
+            for gid in gids_hr:
+                lr_gids = tree.query(hr_res.coordinates[gid])[1]
+                lr_data = lr_res[dset, :, lr_gids]
+                hr_data = mrr[dset, :, gid]
+                assert len(lr_data.shape) == len(hr_data.shape)
+                assert np.allclose(lr_data, hr_data[::12])
```

Comment:
can we define this step size as a global var at the top. didn't immediately know where this 12 came from.

Reply:
can do